light-client/src/supervisor.rs

```diff
@@ -142,6 +133,17 @@ impl Supervisor {
         }
     }
 
+    /// Create a new handle to this supervisor.
+    pub fn handle(&mut self) -> Handle {
+        Handle::new(self.sender.clone())
+    }
+
+    #[pre(self.peers.primary().is_some())]
```

Comment:
I think we should remove this precondition (and other similar ones on `verify_to_highest/target`) since we are actually raising a proper error if there is no primary, but this can be done in a follow-up (eg. by me).

Reply:
Tracking this in #400